### PR TITLE
Corrected keywords for intrinsic functions

### PR DESCRIPTION
**Pull request by Kishan-Ved**
Fixes: https://github.com/lfortran/lfortran/issues/3076
Credit/Reference: https://github.com/lfortran/lfortran/pull/3105

### CODE REVIEW

**Comment by certik** [comment]
Thanks!

Can you please add tests to ensure this change actually works?

Here is the design that we need to move to: https://github.com/lfortran/lfortran/issues/3090

**Comment by Kishan-Ved** [comment]
Sure, I'll start working on it. Thanks!

File: integration_tests/intrinsics_112.f90 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+program MaxExample
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_112.f90:1]:
```suggestion
program intrinsics_112
```

File: integration_tests/intrinsics_112.f90 [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  integer :: a, b, result
+  a = 10
+  b = 7
+  if ( .not. max0(a1=a, a2=b) == 10) error stop
+end program MaxExample
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_112.f90:6]:
```suggestion
end program intrinsics_112
```

File: integration_tests/intrinsics_108.f90 [hunk 3]
```diff
@@ -0,0 +1,5 @@
+program intrinsics_108
+    integer :: x(5) = (/ 1, 2, 3, 4 ,5 /)
+    logical :: mask(5) = (/ .TRUE., .FALSE., .TRUE., .FALSE., .TRUE. /)
+    if ( .not. product(array=x,mask=mask) == 15) error stop
```
[the remark this answers is not in the document]
Comment (HarshitaKalani) [reply integration_tests/intrinsics_108.f90:4]: Can you also add more examples here like:
```
if ( .not. product( x, mask = mask ) == 15 ) error stop
```
```
if ( .not. product( x, mask ) == 15 ) error stop
```

And similarly for other examples too.

File: integration_tests/intrinsics_110.f90 [hunk 4]
```diff
@@ -0,0 +1,6 @@
+program MaxExample
+  integer :: a, b, result
+  a = 10
+  b = 7
+  if ( .not. min0(a1=a, a2=b) == 7) error stop
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_110.f90:5]:
```suggestion
  if ( .not. min0( a1 = a, a2 = b ) == 7 ) error stop
```

Keep a space after `(` and before `)` and other operators

**Comment by Kishan-Ved** [comment]
I have corrected the names.

File: integration_tests/intrinsics_107.f90 [hunk 5]
```diff
@@ -0,0 +1,5 @@
+program intrinsics_107
+    integer :: x(5) = (/ 1, 2, 3, 4 ,5 /)
```
Comment (certik) [inline integration_tests/intrinsics_107.f90:2]: Use `[]`:
```suggestion
    integer :: x(5) = [1, 2, 3, 4 ,5]
```
And so on for the other tests.

**Comment by certik** [comment]
Excellent, I think that starts to look good!

**Comment by certik** [comment]
I think this looks great. Can you please resolve the conflicts? Let's see if tests pass.

**Comment by Kishan-Ved** [comment]
There seems to be some problem with the file: intrinsics_107
It was added by someone else, and hence it appeared as a conflict. I even got an error in it while running integration tests locally, and I fixed it. There are no errors integration test errors on my laptop, but the checks fail due to it.

**Comment by Kishan-Ved** [comment]
For file intrinsics_107:
I got the output on my laptop as:
```
          32
          32
ERROR STOP 
```

so I changed 32 to 64. But now the Test on GitHub fails, error:
```
          32
          64
ERROR STOP 
```

**Comment by certik** [comment]
Leave `intrinsics_107` as it was (all tests pass at the CI as well as for me locally), just add your new file with a higher number.

**Comment by HarshitaKalani** [comment]
Hey @Kishan-Ved 
Please confirm if you followed the same steps
In you branch, after doing
```
git pull origin main
```
resolve the conflicts and then do
```
git clean -dfx
./build0.sh && ./build1.sh
```
And then update the test references and finally run the integration tests.

I think, you didn't do `./build0.sh && ./build1.sh` and that's why: https://github.com/lfortran/lfortran/pull/3123#issuecomment-1898824813 issue is coming up.

**Comment by Kishan-Ved** [comment]
> Hey @Kishan-Ved Please confirm if you followed the same steps In you branch, after doing
> 
> ```
> git pull origin main
> ```
> 
> resolve the conflicts and then do
> 
> ```
> git clean -dfx
> ./build0.sh && ./build1.sh
> ```
> 
> And then update the test references and finally run the integration tests.
> 
> I think, you didn't do `./build0.sh && ./build1.sh` and that's why: [#3123 (comment)](https://github.com/lfortran/lfortran/pull/3123#issuecomment-1898824813) issue is coming up.

Will these steps remove my changes?

**Comment by Kishan-Ved** [comment]
> Leave `intrinsics_107` as it was (all tests pass at the CI as well as for me locally), just add your new file with a higher number.

Done.

**Comment by Kishan-Ved** [comment]
> > Hey @Kishan-Ved Please confirm if you followed the same steps In you branch, after doing
> > ```
> > git pull origin main
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > resolve the conflicts and then do
> > ```
> > git clean -dfx
> > ./build0.sh && ./build1.sh
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And then update the test references and finally run the integration tests.
> > I think, you didn't do `./build0.sh && ./build1.sh` and that's why: [#3123 (comment)](https://github.com/lfortran/lfortran/pull/3123#issuecomment-1898824813) issue is coming up.
> 
> Will these steps remove my changes?

But I would like to know this though, how to fix such things without losing my changes?

File: integration_tests/intrinsics_107.f90 [hunk 6]
```diff
@@ -9,4 +9,4 @@ PROGRAM intrinsics_107
     if ( trailz(x) /= 0 ) error stop
     WRITE (*,*) trailz(y)
     if ( trailz(y) /= 0 ) error stop
-END PROGRAM
\ No newline at end of file
+END PROGRAM
```
Comment (certik) [inline integration_tests/intrinsics_107.f90:12]: remove this change

File: integration_tests/CMakeLists.txt [hunk 7]
```diff
@@ -610,7 +610,19 @@ RUN(NAME intrinsics_103 LABELS gfortran llvm)
 RUN(NAME intrinsics_104 LABELS gfortran llvm c wasm)
 RUN(NAME intrinsics_105 LABELS gfortran llvm)#shape
 RUN(NAME intrinsics_106 LABELS gfortran llvm)#merge
-RUN(NAME intrinsics_107 LABELS gfortran llvm)#trailz
+RUN(NAME intrinsics_107 LABELS gfortran llvm)
```
Comment (certik) [inline integration_tests/CMakeLists.txt:613]: Remove this change:
```suggestion
RUN(NAME intrinsics_107 LABELS gfortran llvm)#trailz
```

File: integration_tests/intrinsics_107.f90 (hunk 6 above)
Comment (Kishan-Ved) [reply integration_tests/intrinsics_107.f90:12]: I didn't change this, how do I remove it?

**Comment by Kishan-Ved** [comment]
There is a new file, intrinsics_108, should I change my file to a higher number?

File: integration_tests/intrinsics_107.f90 (hunk 6 above)
Comment (certik) [reply integration_tests/intrinsics_107.f90:12]: Use
```
git checkout main -- integration_tests/intrinsics_107.f90
git commit
```

**Comment by certik** [comment]
> There is a new file, intrinsics_108, should I change my file to a higher number?

Yes.

**Comment by certik** [comment]
See here https://github.com/lcompilers/lpython/blob/30b5bfc41774833fb94ec0c2e1926ce33be10fdb/doc/src/rebasing.md how to use git for these kinds of rebasing and cleanup tasks.

File: integration_tests/intrinsics_107.f90 (hunk 6 above)
Comment (Kishan-Ved) [reply integration_tests/intrinsics_107.f90:12]: I did this, but it says there's no such file found

**Comment by Kishan-Ved** [comment]
I think we should keep x and y in hypot. (According to: https://gcc.gnu.org/onlinedocs/gfortran/HYPOT.html)

File: integration_tests/intrinsics_107.f90 (hunk 6 above)
Comment (certik) [reply integration_tests/intrinsics_107.f90:12]: Do `git fetch origin` first. Ensure `origin` points to `https://github.com/lfortran/lfortran`.

Comment (Kishan-Ved) [reply integration_tests/intrinsics_107.f90:12]: Sir I tried that, I still get the error:
`(lf) kishan@kishan-IdeaPad-3-15IIL05:~/Desktop/lfortran$ git checkout main -- integration_tests/intrinsics_107.f90
error: pathspec 'integration_tests/intrinsics_107.f90' did not match any file(s) known to git`

Comment (certik) [reply integration_tests/intrinsics_107.f90:12]: It's right here: https://github.com/lfortran/lfortran/blob/main/integration_tests/intrinsics_107.f90, so you need to figure out why you don't have it in your local git. You might be in a wrong branch, wrong directory, didn't checkout the latest main, and so on.

Comment (Kishan-Ved) [reply integration_tests/intrinsics_107.f90]: Sir I figured this out 👍🏻, I got to learn a new thing today 🙂. Thank you.

File: integration_tests/intrinsics_118.f90 [hunk 8]
```diff
@@ -0,0 +1,3 @@
+program intrinsics_118
+    print *, hypot( x = 1.e0_4, y = 0.5e0_4 )
+end program intrinsics_118
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_118.f90:3]: You can add error stop conditions in this file too.

File: integration_tests/intrinsics_120.f90 [hunk 9]
```diff
@@ -0,0 +1,7 @@
+program intrinsics_120
+    integer :: x(5) = [ 1, 2, 3, 4 ,5 ]
+    logical :: mask(5) = [ .TRUE., .FALSE., .TRUE., .FALSE., .TRUE. ]
+    if ( .not. product( array = x, mask = mask ) == 15 ) error stop
+    if ( .not. product( x, mask = mask ) == 15 ) error stop
+    if ( .not. product( x, mask ) == 15 ) error stop
+end
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_120.f90:7]: We can follow this for other examples too to ensure that this approach works for all the cases - when we're putting some parameters as arguments directly (line 6) and while some are inserted as keywords(line 4 an line 5).

**Review by HarshitaKalani — COMMENTED** [review]
This looks good to me. I just left some comments for improvement.

**Review by certik — APPROVED** [review]

**Comment by certik** [comment]
Thanks for finishing the PR!

File: integration_tests/intrinsics_118.f90 (hunk 8 above)
Comment (certik) [reply integration_tests/intrinsics_118.f90:3]: @Kishan-Ved please submit a new PR with this improvement. I already set this one to merge.

Comment (Kishan-Ved) [reply integration_tests/intrinsics_118.f90:3]: Ok Sir. I'll do it.

**Pull request merged**